From 704358b86df776d03cca9f36c8d1ec69b23a350f Mon Sep 17 00:00:00 2001
From: Kent Fredric <kentnl@gentoo.org>
Date: Wed, 22 Jan 2020 19:43:35 +1300
Subject: Fix cargo for stripped targets

- Remove references to targets stripped from the image
- Remove dev-dependencies implicated by only stripped targets
---
 Cargo.toml | 132 -----------------------------------------------------
 1 file changed, 132 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index ae9b68c..f53b013 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,120 +34,6 @@ incremental = false
 codegen-units = 1
 incremental = false
 
-[[example]]
-name = "client"
-path = "examples/client.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "client_json"
-path = "examples/client_json.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "echo"
-path = "examples/echo.rs"
-required-features = ["runtime", "stream"]
-
-[[example]]
-name = "gateway"
-path = "examples/gateway.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "hello"
-path = "examples/hello.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "http_proxy"
-path = "examples/http_proxy.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "multi_server"
-path = "examples/multi_server.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "params"
-path = "examples/params.rs"
-required-features = ["runtime", "stream"]
-
-[[example]]
-name = "send_file"
-path = "examples/send_file.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "single_threaded"
-path = "examples/single_threaded.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "state"
-path = "examples/state.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "tower_client"
-path = "examples/tower_client.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "tower_server"
-path = "examples/tower_server.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "upgrades"
-path = "examples/upgrades.rs"
-required-features = ["runtime"]
-
-[[example]]
-name = "web_api"
-path = "examples/web_api.rs"
-required-features = ["runtime", "stream"]
-
-[[test]]
-name = "client"
-path = "tests/client.rs"
-required-features = ["runtime", "stream"]
-
-[[test]]
-name = "integration"
-path = "tests/integration.rs"
-required-features = ["runtime", "stream"]
-
-[[test]]
-name = "server"
-path = "tests/server.rs"
-required-features = ["runtime"]
-
-[[bench]]
-name = "body"
-path = "benches/body.rs"
-required-features = ["runtime", "stream"]
-
-[[bench]]
-name = "connect"
-path = "benches/connect.rs"
-required-features = ["runtime"]
-
-[[bench]]
-name = "end_to_end"
-path = "benches/end_to_end.rs"
-required-features = ["runtime"]
-
-[[bench]]
-name = "pipeline"
-path = "benches/pipeline.rs"
-required-features = ["runtime"]
-
-[[bench]]
-name = "server"
-path = "benches/server.rs"
-required-features = ["runtime", "stream"]
 [dependencies.bytes]
 version = "0.5"
 
@@ -204,27 +90,9 @@ version = "0.3"
 features = ["alloc"]
 default-features = false
 
-[dev-dependencies.matches]
-version = "0.1"
-
-[dev-dependencies.num_cpus]
-version = "1.0"
-
 [dev-dependencies.pretty_env_logger]
 version = "0.4"
 
-[dev-dependencies.serde]
-version = "1.0"
-
-[dev-dependencies.serde_derive]
-version = "1.0"
-
-[dev-dependencies.serde_json]
-version = "1.0"
-
-[dev-dependencies.spmc]
-version = "0.3"
-
 [dev-dependencies.tokio]
 version = "0.2.2"
 features = ["fs", "macros", "io-std", "rt-util", "sync", "time", "test-util"]
-- 
2.25.0

